src/tensor_parallel/slicing_configs.py

```diff
@@ -133,17 +137,23 @@ def select_kv_for_rank(*kvs, rank):
     return Config(
         state_rules={
             # T5Attention
-            r".*SelfAttention\.q\.(weight|bias)$": partial(
-                split_heads, dim=0, head_dim=head_dim, world_size=world_size
+            r".*SelfAttention\.q\.(weight|bias)$": (
+                partial(split_heads, dim=0, head_dim=head_dim, world_size=world_size),
+                "split 0",
```

Comment:
__Nitpick:__ it sounds a bit counterintuitive that I have to specify "split 0" instead of something like "gather 0" or "inv_split 0"as the inverse action.

Perhaps we can make this kind of behavior:
 - if there's one string, do the pre-recorded inverse
 - if there are two, the second action specifies the inverse transform

Again, this is a nitpick, feel free to ignore

Reply:
I'm currently working on a config refactor that replaces those strings and callable with specific interfaces.
I'll think of a better solution in #43